src/main/java/org/jenkinsci/test/acceptance/junit/JenkinsAcceptanceTestRule.java

```diff
@@ -110,7 +110,12 @@ public int compare(Integer o1, Integer o2) {
 
                 for (Set<TestRule> rulesGroup: rules.values()) {
                     for (TestRule rule: rulesGroup) {
-                        body = rule.apply(body, description);
+                        try {
+                            body = rule.apply(body, description);
+                        } catch (Exception e) { //This is to note which rule is failing when a rule does fail as the evaluate method hinds this information
+                            System.out.println("Test Rule " + rule.getClass() + " failed to apply. Failed with: " + e.getCause());
+                            throw e;
```

Comment:
I would be better to wrap to another exception indicating the rule class in message as this way the stack trace and the rule name are both reported but by separate mechanisms.

Reply:
That was my original plan but I didn't want to change the exception type as that was what is sent to the controller in the diagnose method. I can change it to a warped exception.